src/io/input_split_base.cc

```diff
@@ -191,7 +193,9 @@ size_t InputSplitBase::Read(void *ptr, size_t size) {
     offset_curr_ += n;
     if (nleft == 0) break;
     if (n == 0) {
-      buf[0] = '\n'; ++buf; --nleft;  // insert a newline between files
+      if (is_text_parser) {
+        buf[0] = '\n'; ++buf; --nleft;  // insert a newline between files
```

Comment:
can you write a comment, why are we inserting \n for text parser

Reply:
I'll add a link to the issue that gave rise to it.